$OpenBSD: patch-configure_in,v 1.3 2014/05/09 11:40:12 ajacoutot Exp $
Add OpenBSD default configuration profile.
Force using of GNU flex.
Help to find libwbxml.
--- configure.in.orig	Thu Aug  2 02:03:31 2012
+++ configure.in	Tue Jan 29 23:07:18 2013
@@ -234,6 +234,23 @@ case ${VIRTUOSO_LAYOUT} in
 	hostingdir=$libdir/virtuoso/hosting
 	;;
 
+  [[Op]][[Pp]][[Ee]][[Nn]][[Bb]][[Ss]][[Dd]])		# OpenBSD
+	virtuoso_layout=OpenBSD
+	if test "x$prefix" = "xNONE"
+	then
+		prefix=${PREFIX}
+		localstatedir=${LOCALSTATEDIR}
+	fi
+	vaddir=$datadir/virtuoso/vad
+	docdir=$datadir/doc/virtuoso
+	htmldocdir=$datadir/doc/html/virtuoso
+	pdfdocdir=$datadir/doc/virtuoso/pdf
+	demodir=$localstatedir/virtuoso/demo
+	dbdir=$localstatedir/virtuoso/db
+	httprootdir=$localstatedir/virtuoso/vsp
+	hostingdir=$libdir/virtuoso/hosting
+	;;
+
   [[Oo]][[Pp]][[Tt]])					# /opt
 	virtuoso_layout=opt
 	test "x$prefix" = xNONE && prefix=/opt/virtuoso-opensource
@@ -263,7 +280,7 @@ case ${VIRTUOSO_LAYOUT} in
   *)							# UNKNOWN
   	AC_MSG_RESULT(unknown)
 	echo ""
-	echo "AC_HELP_STRING([Supported layouts:], [Debian FreeBSD GNU OpenLink opt RedHat])"
+	echo "AC_HELP_STRING([Supported layouts:], [Debian FreeBSD OpenBSD GNU OpenLink opt RedHat])"
 	echo ""
 	AC_MSG_ERROR([unable to continue])
 	;;
@@ -383,9 +400,10 @@ AC_PROG_YACC
 AC_PATH_PROG(HTMLDOC, htmldoc)
 AM_CONDITIONAL(WITH_HTMLDOC, test "x$HTMLDOC" != "x")
 
-AM_PROG_LEX
+# AM_PROG_LEX
 
-AC_PATH_PROG(FLEX, flex, flex)
+AC_PATH_PROG(LEX, gflex, gflex)
+AC_PATH_PROG(FLEX, gflex, gflex)
 AC_PATH_PROG(GAWK, gawk, gawk)
 AC_PATH_PROG(GPERF, gperf, gperf)
 AC_PATH_PROG(PERL, perl, perl)
@@ -1403,7 +1421,7 @@ then
 	   WBXML2_LDFLAGS="-lwbxml2"
 	   ;;
 	*)
-	   WBXML2_CFLAGS="-I$with_wbxml2/include"
+	   WBXML2_CFLAGS="-I$with_wbxml2/include/wbxml"
 	   WBXML2_LDFLAGS="-L$with_wbxml2/lib -lwbxml2"
 	   LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$with_wbxml2/lib"
 	   export LD_LIBRARY_PATH
@@ -1421,7 +1439,8 @@ then
 int
 main ()
 {
-  WBXMLParser *wbxml_parser = wbxml_parser_create();
+  WB_UTINY *wbxml = wbxml_realloc(NULL, 100);
+  wbxml_free(wbxml);
 
   return 0;
 }],
